src/Compilers/CSharp/Portable/Binder/SwitchBinder_Patterns.cs

```diff
@@ -218,6 +218,8 @@ internal override void BindPatternSwitchLabelForInference(CasePatternSwitchLabel
                         var caseLabelSyntax = (CaseSwitchLabelSyntax)node;
                         BoundConstantPattern pattern = sectionBinder.BindConstantPattern(
                             node, SwitchGoverningType, caseLabelSyntax.Value, node.HasErrors, diagnostics, out bool wasExpression);
+                        if (!pattern.HasErrors && caseLabelSyntax.Value is IdentifierNameSyntax name && name.Identifier.ContextualKind() == SyntaxKind.UnderscoreToken)
+                            diagnostics.Add(ErrorCode.WRN_CaseConstantNamedUnderscore, caseLabelSyntax.Value.Location);
                         pattern.WasCompilerGenerated = true; // we don't have a pattern syntax here
                         bool hasErrors = pattern.HasErrors;
```

Comment:
>bool hasErrors = pattern.HasErrors; [](start = 24, length = 35)

Consider moving above `if` and using `hasErrors` there.

Reply:
Not applicable after extracting local function.

---
In reply to: [230161807](https://github.com/dotnet/roslyn/pull/30847#discussion_r230161807) [](ancestors = 230161807)